### PR TITLE
CMS Reporting UI

### PR DESCRIPTION
**Pull request by Stanton**

### CODE REVIEW

File: stylesheets/_layout.report.scss [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        background-color: color(gray, off-white);
+    }
+
+    .tabs__content {
+        padding: 0;
+    }
+
+    .breadcrumb,
+    .main-title {
+        background-color: white;
```
Comment (Stanton) [inline stylesheets/_layout.report.scss]: Use palette

File: stylesheets/_layout.report.scss [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        padding-bottom: 11px;
+    }
+
+    .main-title {
+        margin-bottom: 0;
+        padding-bottom: 13px;
+    }
+
+    .nav-inline {
+        background-color: white;
```
Comment (Stanton) [inline stylesheets/_layout.report.scss]: Use palette

File: stylesheets/pulsar.scss [hunk 3]
```diff
@@ -64,6 +64,7 @@ $FontAwesomePath: '../libs/font-awesome/font/';
 @import 'layout.sidebar';
 @import 'layout.signin';
 @import 'layout.tabs';
+@import 'layout.report';
```
Comment (Stanton) [inline stylesheets/pulsar.scss]: Alphabetise?

File: js/index.js [hunk 4]
```diff
@@ -19,10 +19,12 @@ var $                     = require('jquery'),
     matchMedia            = require('./polyfills/matchMedia'),
     matchMediaAddListener = require('./polyfills/matchMedia.addListener');
 
+	chartist  = require('../libs/chartist/dist/chartist.min'),
```
Comment (Stanton) [inline js/index.js]: indentation

File: bower.json [hunk 5]
```diff
@@ -31,7 +31,9 @@
     "susy": "~2.2.6",
     "svgeezy": "latest",
     "Vague.js": "0.0.4",
-    "vide": "~0.3.1"
+    "vide": "~0.3.1",
+    "chartist": "^0.9.8",
+    "bootstrap-daterangepicker": "latest"
```
Comment (Stanton) [inline bower.json]: lock to specific version

File: js/index.js [hunk 6]
```diff
@@ -19,10 +19,12 @@ var $                     = require('jquery'),
...
     clickover = require('../libs/bootstrapx-clickover/js/bootstrapx-clickover'),
     history   = require('../libs/history.js/scripts/bundled/html5/jquery.history'),
     jqueryui  = require('../libs/jquery-ui/jquery-ui.min'),
     countdown = require('../libs/jquery.countdown/dist/jquery.countdown.min'),
+    daterangepicker = require('../libs/bootstrap-daterangepicker/daterangepicker'),
```
Comment (Stanton) [inline js/index.js]: alphabetise

File: js/PulsarFormComponent.js [hunk 7]
```diff
@@ -40,7 +40,7 @@ PulsarFormComponent.prototype.init = function () {
     });
 
     // Block styled checkboxes and radios
-    var choiceBlock = component.$html.find(".choice--block");
+    var choiceBlock = component.$html.find(".choice--block, .choice--bubbles");
```
Comment (jamesjacobs) [inline js/PulsarFormComponent.js]: variable declarations should be at the top of the function

File: js/main.js [hunk 8]
```diff
@@ -65,6 +65,62 @@
 
         tinycon.setBubble(6);
 
-    });
 
+        new chartist.Line('.ct-chart', {
```
Comment (jamesjacobs) [inline js/main.js]: Could we move the chartist code to it's own class/module?

File: stylesheets/_layout.report.scss [hunk 9]
```diff
@@ -75,6 +75,11 @@
     text-align: right;
 }
 
+.report__chart {
+    margin-bottom: 24px;
+    height: 240px;
```
Comment (jamesjacobs) [inline stylesheets/_layout.report.scss]: Alphabetize order

File: js/main.js [hunk 10]
```diff
@@ -51,6 +51,19 @@
                 $(this).text('Menu');
             }
         });
+
+        new chartist.Line('.ct-chart', {
```
Comment (jamesjacobs) [inline js/main.js]: Consider moving to charist module

File: stylesheets/_component.choice.scss [hunk 11]
```diff
@@ -43,6 +43,34 @@
     }
 }
 
+.choice--bubbles > .controls {
+    .control__label {
+        background-color: color(gray, off-white);
+        border-radius: 200px;
+        cursor: pointer;
+        color: color(gray, light);
```
Comment (jamesjacobs) [inline stylesheets/_component.choice.scss]: alphabetize

File: stylesheets/_component.choice.scss [hunk 12]
```diff
@@ -51,7 +79,8 @@
     }
 }
 
-.choice--block-inline {
+.choice--block-inline,
+.choice--bubbles-inline {
     .controls {
         display: flex;
```
Comment (jamesjacobs) [inline stylesheets/_component.choice.scss]: Do we have a flex fallback for old IE?

File: stylesheets/_component.dropdowns.scss [hunk 13]
```diff
@@ -35,14 +35,14 @@
...
 	font-size: $dropdown-font-size;
 	min-width: 160px;
-	padding: 5px 0;
 	margin: 0; // override default ul
+	padding: 5px 0;
 	list-style: none;
 	background-color: $dropdown-bg;
-	border: 1px solid $dropdown-fallback-border; // IE8 fallback
-	border: 1px solid $dropdown-border;
+	// border: 1px solid $dropdown-fallback-border; // IE8 fallback
```
Comment (jamesjacobs) [inline stylesheets/_component.dropdowns.scss]: Can we remove these commented out lines?

File: stylesheets/_component.dropdowns.scss [hunk 14]
```diff
@@ -35,14 +35,14 @@
...
+	// border: 1px solid $dropdown-border;
 	// border-radius: $border-radius;
-	box-shadow: 0 6px 12px rgba(0, 0, 0, .175);
+	box-shadow: 0 6px 12px rgba(0, 0, 0, .3);
```
Comment (jamesjacobs) [inline stylesheets/_component.dropdowns.scss]: alphabetize

File: stylesheets/_component.dropdowns.scss [hunk 15]
```diff
@@ -299,3 +299,26 @@
...
+
+.dropdown__title {
+	border-bottom: 1px solid color(border);
+	line-height: 48px;
+	margin: -5px 0 0;
+	padding: 0 50px 0 20px;
+
+	> .btn {
+		position: absolute;
+		top: 8px;
```
Comment (jamesjacobs) [inline stylesheets/_component.dropdowns.scss]: alphabetize

File: views/playground/reporting/report.html.twig [hunk 16]
```diff
@@ -76,12 +96,23 @@
...
+                })
+            }}
         </div>
 
         <h1 class="report__title">All content owned by Paul Stanton</h1>
         <span class="report__info">56 items found &mdash; Last run 01/08/2016 12:34pm by Paul Stanton</span>
 
         <div class="report__data">
+
+            <div class="report__chart ct-chart"></div>
```
Comment (jamesjacobs) [inline views/playground/reporting/report.html.twig]: Are we hiding the chart if no-js?

File: stylesheets/_layout.tabs.scss [hunk 17]
```diff
@@ -5,6 +5,7 @@
 .content-main {
```
Comment (jamesjacobs) [inline stylesheets/_layout.tabs.scss]: alphabetize styles

File: stylesheets/_layout.report.scss [hunk 18]
```diff
@@ -95,13 +64,13 @@
 }
 
 .report__chart {
-    margin-bottom: 24px;
+    margin-bottom: $line-height-base;
```
Comment (jamesjacobs) [inline stylesheets/_layout.report.scss]: alphabetize

File: stylesheets/_layout.report.scss [hunk 19]
```diff
@@ -169,10 +138,10 @@
     top: -30px;
     background-color: color(background, light);
     box-shadow: 0 6px 12px rgba(0, 0, 0, .3);
-    border-top-left-radius: 20px;
-    border-bottom-right-radius: 20px;
+    border-top-left-radius: $border-radius-large;
+    border-bottom-right-radius: $border-radius-large;
```
Comment (jamesjacobs) [inline stylesheets/_layout.report.scss]: alphabetize

**Comment by jamesjacobs** [comment]
Tried to review, GH had issues due to 326 files in the diff 😓

File: views/playground/reporting/report.html.twig (hunk 16 above)
Comment (Stanton) [inline views/playground/reporting/report.html.twig]: The container wouldn't be populated if js is not available

**Pull request merged**